codex/rest/api.nim

```diff
@@ -110,6 +110,20 @@ proc retrieveCid(
 proc initDataApi(node: CodexNodeRef, repoStore: RepoStore, router: var RestRouter) =
   let allowedOrigin = router.allowedOrigin # prevents capture inside of api defintion
 
+  router.api(
+    MethodOptions,
+    "/api/codex/v1/data") do (
+       resp: HttpResponseRef) -> RestApiResponse:
+
+      if corsOrigin =? allowedOrigin:
+        resp.setHeader("Access-Control-Allow-Origin", corsOrigin)
+        resp.setHeader("Access-Control-Allow-Methods", "POST, OPTIONS")
+        resp.setHeader("Access-Control-Headers", "X-Requested-With")
+        resp.setHeader("Access-Control-Allow-Headers", "content-type")
```

Comment:
We may also consider directing the browser to [cache the pre-flight response](https://developer.mozilla.org/en-US/docs/Glossary/Preflight_request) so it doesn't need to continually re-issue it for some time:
> The preflight response can be optionally cached for the requests created in the same [URL](https://developer.mozilla.org/en-US/docs/Glossary/URL) using [Access-Control-Max-Age](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Max-Age) header like in the above example. To cache preflight responses, the browser uses a specific cache that is separate from the general HTTP cache that the browser manages. Preflight responses are never cached in the browser's general HTTP cache.

Reply:
Good catch!

Reply:
Yes Good catch, I added the "Access-Control-Max-Age" header.